source/Halibut/Transport/Streams/StreamFactory.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.IO;
+using System.Net.Sockets;
+using System.Net.WebSockets;
+using Halibut.Transport.Protocol;
+using Halibut.Util;
+
+namespace Halibut.Transport.Streams
+{
+    public class StreamFactory : IStreamFactory
+    {
+        public StreamFactory(AsyncHalibutFeature asyncHalibutFeature)
+        {
+            AsyncHalibutFeature = asyncHalibutFeature;
+        }
+
+        public AsyncHalibutFeature AsyncHalibutFeature { get; }
```

Comment:
Nit: My default is to make things private readonly unless needed. The lower the public surface, the simpler things are 😁. I don't think we need this exposed?